internals/overlord/overlord.go

```diff
@@ -64,21 +64,42 @@ type Overlord struct {
 	pruneTicker *time.Ticker
 
 	// managers
-	inited     bool
-	runner     *state.TaskRunner
-	serviceMgr *servstate.ServiceManager
-	commandMgr *cmdstate.CommandManager
-	checkMgr   *checkstate.CheckManager
-	logMgr     *logstate.LogManager
+	inited           bool
+	runner           *state.TaskRunner
+	serviceMgr       *servstate.ServiceManager
+	commandMgr       *cmdstate.CommandManager
+	checkMgr         *checkstate.CheckManager
+	logMgr           *logstate.LogManager
+	externalManagers map[any]StateManager
 }
 
-// New creates a new Overlord with all its state managers.
+// ManagerProvider is the interface that ManagerGenerator depends on
+//
+// Overlord implements ManagerProvider as it provides the necessary
+// handles to hook an external manager into the overlord's environment.
+type ManagerProvider interface {
+	State() *state.State
+	TaskRunner() *state.TaskRunner
+}
+
+// ManagerGenerator is passed to Overlord to create a manager
+// The return value is a key, value pair where the key has to be a unique
+// identifier for the manager being created.
+type ManagerGenerator func(ManagerProvider) (key any, manager StateManager)
```

Comment:
I think we will have to include returning an error to comply with how each manager instantiation can fail, and how each of these errors are propagated to the ```overlord.New()``` error return.